cumulusci/utils/fileutils.py

```diff
@@ -86,6 +92,206 @@ def load_from_source(
             yield path, f
 
 
+def cleanup_org_cache_dirs(keychain, project_config):
+    """Cleanup directories that are not associated with a connected/live org."""
+    domains = set()
+    for org in keychain.list_orgs():
+        org_config = keychain.get_org(org)
+        domain = org_config.get_domain()
+        if domain:
+            domains.add(domain)
+
+    assert project_config.project_cache_dir
+    assert keychain.global_config_dir
```

Comment:
These could be the same when running `cci org list` outside of a project directory, right? So probably should add the glob results to a set to avoid processing twice.

Reply:
Also I'd be inclined to put this function in cumulusci.core somewhere. cumulusci.utils mostly has standalone util functions that don't have a dependency on cumulusci domain objects

Reply:
Moved. 216b7482

Also renamed BaseProjectConfig.py so I could do mocking.

Reply:
You mean BaseProjectKeychain. BaseProjectConfig was renamed to project_config quite a while ago.